tools/slicec-cs/src/generators/class_generator.rs

```diff
@@ -70,41 +70,39 @@ pub fn generate_class(class_def: &Class) -> CodeBlock {
 
     let constructor_summary = format!(r#"Constructs a new instance of <see cref="{class_name}" />."#);
 
-    // The primary constructor (may be parameterless)
-    class_builder.add_block(constructor(
-        &class_name,
-        access,
-        constructor_summary.clone(),
-        &namespace,
-        &fields,
-        &base_fields,
-    ));
-
-    // Secondary constructor for all fields minus those with optional types.
-    // This constructor is only generated if necessary
-    if non_nullable_fields.len() + non_nullable_base_fields.len() < fields.len() + base_fields.len() {
+    if !class_def.all_fields().is_empty() {
+        // parameterless constructor.
+        // The constructor needs to be public for System.Activator.CreateInstance.
+        let mut parameterless_constructor = FunctionBuilder::new("public", "", &class_name, FunctionType::BlockBody);
+        parameterless_constructor.add_comment("summary", constructor_summary.clone());
+        class_builder.add_block(parameterless_constructor.build());
+
+        // The primary constructor.
         class_builder.add_block(constructor(
             &class_name,
             access,
-            constructor_summary,
+            constructor_summary.clone(),
             &namespace,
-            &non_nullable_fields,
-            &non_nullable_base_fields,
+            &fields,
+            &base_fields,
         ));
-    }
 
-    // parameterless constructor for decoding, generated only if the preceding constructors are not parameterless
-    if non_nullable_fields.len() + non_nullable_base_fields.len() > 0 {
-        // The constructor needs to be public for System.Activator.CreateInstance.
-        let mut decode_constructor = FunctionBuilder::new("public", "", &class_name, FunctionType::BlockBody);
-        decode_constructor.add_never_editor_browsable_attribute();
-        decode_constructor.add_comment(
-            "summary",
-            format!(r#"Constructs a new instance of <see cref="{class_name}"/> for the Slice decoder."#),
-        );
-        decode_constructor.set_body(initialize_required_fields(&fields));
-        class_builder.add_block(decode_constructor.build());
+        // Secondary constructor for all fields minus those with optional types.
```

Comment:
The comment should use "nullable" for consistency with terminology in the code below.